Log underlying error when YAML generation fails

ejweber opened the pull request:
Fixes #99 with best-effort.

I can't actually manage to hit an error like the on from https://github.com/rancher/support-bundle-kit/issues/99#issue-2215521652 in my cluster. I tried mangling the `cluster-admin` ClusterRole to remove access to nonResourceURLs and removing the `system:discovery` ClusterRoleBinding in an attempt to stop support-bundle-manager from using the Kubernetes Discovery API. This DID result in a support bundle without YAML files, but did not trigger the `Unable to fetch cluster resources` message (in either a modified or unmodified support-bundle-kit). This means I cannot actually show the result of this PR in action. However, it should provide us additional debugging information if the YAML collection fails again like this in the future.

Vicente-Cheng reviewed (approved):
Thanks! We need this one.

We also get some SBs w/o yaml files and cannot find anything from the SB generator log.

Vicente-Cheng commented:
@ejweber, could you help to rebase for the conflict?

ejweber commented:
It looks like @ibrokethecloud beat me to some of these locations, but I think these errors also make sense in the error log, so we should move forward with the changes IMO. Rebased and fixed conflicts.

bk201 reviewed (approved):
Thanks, Eric.
